cpp/src/arrow/engine/substrait/extension_set.h

```diff
@@ -224,7 +240,7 @@ class ARROW_ENGINE_EXPORT ExtensionSet {
   std::size_t num_functions() const { return functions_.size(); }
 
  private:
-  ExtensionIdRegistry* registry_;
+  const ExtensionIdRegistry* registry_;
```

Comment:
Is there any particular reason to change this to a `const`?  I think it's fine but it seems unrelated to the change so I wanted to make sure I wasn't missing anything.